### PR TITLE
Add docs to package.json scripts and remove / fix warnings

### PR DESCRIPTION
**Pull request by rmjohnson**
Addresses #29

### CODE REVIEW

File: src/Async.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@
  *   never started, no work is performed.)
  *
  * This API is curreid and has been optimized for use right-to-left
- * function composition like {@link pipe} or {@link flow}.
+ * function composition like pipe or flow.
```
Comment (taylorn16) [inline src/Async.ts]: ❓ would these `@link`s work if we imported pipe and flow at the top of this file?

File: src/Option.ts [hunk 2]
```diff
@@ -341,7 +341,7 @@ export const map3 =
  * Constructs an `Option` from a potentially nullish value.
  * Nullish values will result in a `None` instance. Other
  * values will result in a `Some` instance containing the
- * value now constrained to be {@link NonNullable}.
+ * value now constrained to be NonNullable.
```
Comment (taylorn16) [inline src/Option.ts]:
```suggestion
 * value now constrained to be `NonNullable`.
```

File: package.json [hunk 3]
```diff
@@ -16,7 +16,8 @@
         "test": "vitest",
         "test:coverage": "vitest --watch=false --coverage",
         "lint": "eslint . --ext .ts",
-        "fmt": "prettier . --write"
+        "fmt": "prettier . --write",
```
Comment (taylorn16) [inline package.json:19]: ❓ should we add `docs` to the `.prettierignore` file?

Comment (rmjohnson) [reply package.json:19]: good call :+1:

File: src/Async.ts (hunk 1 above)
Comment (rmjohnson) [reply src/Async.ts]: Just tried several combinations of different links / imports and couldn't get any of them to work. =( let me know if you figure it out and we can update.

File: package.json (hunk 3 above)
Comment (rmjohnson) [reply package.json:19]: https://github.com/fp-toolkit/fp-toolkit/pull/33/commits/bb3e5d2bce5baf74a293fc83a2c069d53b9decb3

File: src/Async.ts (hunk 1 above)
Comment (rmjohnson) [reply src/Async.ts]: Got it working with Nick's help :grin: for reference: https://typedoc.org/guides/declaration-references/

https://github.com/fp-toolkit/fp-toolkit/pull/33/commits/25a048b0230f6983f9766431b286b35ee5ae10ac

**Comment by rmjohnson** [comment]
@taylorn16 I think this is ready for further review when you get a chance. =)

File: src/Async.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /**
- * The `Async` type reperesents a "lazy" or "cold" asynchronous
```
Comment (taylorn16) [inline src/Async.ts:2]: lol nice catch :back-away:

File: src/Async.ts [hunk 5]
```diff
@@ -142,7 +142,7 @@ export const flatten =
 export const unit: Async<{}> = of({})
 
 /**
- * Adds an abitrary delay to an `Async` computation.
+ * Adds an arbitrary delay to an `Async` computation.
```
Comment (taylorn16) [inline src/Async.ts:145]: man was I drunk when I was writing this??

**Review by taylorn16 — APPROVED** [review]
Looks great. Nice work, @rmjohnson ! 🚀 🏁 ✔️

File: src/Async.ts (hunk 4 above)
Comment (rmjohnson) [reply src/Async.ts:2]: Yay for the spell checker VS code extension :joy:

**Pull request merged**